docs/flow/reference/components/prometheus.remote_write.md

```diff
@@ -0,0 +1,357 @@
+inherited from Go's default minimum version, TLS 1.2. If `min_version` is
+provided, it must be set to one of the following strings:
+
+* `"TLS10"` (TLS 1.0)
+* `"TLS11"` (TLS 1.1)
+* `"TLS12"` (TLS 1.2)
+* `"TLS13"` (TLS 1.3)
+
+### Block: `endpoint` > `http_client_config` > `tls_config`
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`ca_file` | `string` | CA certificate to validate the server with. | | no
+`cert_file` | `string` | Certificate file for client authentication. | | no
+`key_file` | `string` | Key file for client authentication. | | no
+`server_name` | `string` | ServerName extension to indicate the name of the server. | | no
+`insecure_skip_verify` | `bool` | Disables validation of the server certificate. | | no
+`min_version` | `string` | Minimum acceptable TLS version. | | no
+
+When `min_version` is not provided, the minumum acceptable TLS version is
+inherited from Go's default minimum version, TLS 1.2. If `min_version` is
+provided, it must be set to one of the following strings:
+
+* `"TLS10"` (TLS 1.0)
+* `"TLS11"` (TLS 1.1)
+* `"TLS12"` (TLS 1.2)
+* `"TLS13"` (TLS 1.3)
+
+### Block: `endpoint` > `queue_config`
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`capacity` | `number` | Number of samples to buffer per shard. | `2500` | no
+`min_shards` | `number` | Minimum amount of concurrent shards sending samples to the endpoint. | `1` | no
+`max_shards` | `number` | Maximum number of concurrent shards sending samples to the endpoint. | `200` | no
+`max_samples_per_send` | `number` | Maximum number of samples per send. | `500` | no
+`batch_send_deadline` | `duration` | Maximum time samples will wait in the buffer before sending. | `"5s"` | no
+`min_backoff` | `duration` | Initial retry delay. The backoff time gets doubled for each retry. | `"30ms"` | no
+`max_backoff` | `duration` | Maximum retry delay. | `"5s"` | no
+`retry_on_http_429` | `bool` | Retry when an HTTP 429 status code is received. | `false` | no
+
+Each queue then manages a number of concurrent _shards_ which is responsible
+for sending a fraction of data to their respective endpoints. The number of
+shards is automatically raised if samples are not being sent to the endpoint
+quickly enough. The range of permitted shards can be configured with the
+`min_shards` and `max_shards` arguments.
+
+Each shard has a buffer of samples it will keep in memory, controlled with the
+`capacity` argument. New metrics won't be read from the WAL unless there is at
+least one shard that is not at maximum capcity.
+
+The buffer of a shard will be flushed and sent to the endpoint either after the
+shard reaches the number of samples specified by `max_samples_per_send` or the
+duration specified by `batch_send_deadline` has elapsed since the last flush
+for that shard.
+
+Shards will infinitely retry requests which fail with an `HTTP 5xx` status code
```

Comment:
Do we want to mention that retry_on_http_429 will honor the `Retry-After`header if specified?

Reply:
TIL :) Definitely, I'll include that